### ISSUE TITLE
No source flattening option?

### ISSUE THREAD

**Issue by tvanier** [lukeed/taskr#103]
I am exploring Fly, looks very interesting. Trying to write a `copy_images` task to recursively copy an `images` folder, seems `Fly.prototype.source` flattens by default, is there a special reason for this? Would a "no flatten" option make sense? Or better to use a separate package like [ncp](https://www.npmjs.com/package/ncp) to copy folders?

``` javascript
export function* copy_images () {
    yield this.source('images/**/*')
        .target('dist/');
}
```

**Comment by ghost** [comment]
`Fly.prototype.source` will flatten its arguments, but it will do nothing to the expanded globs. Perhaps this is a bug?

A `no-flatten` option would make sense and using `ncp` sounds good as well assuming this is not a bug and the expected behavior is actually what you need.

`this.source('images/**/*')` should expand to an array of everything under `images`.

Please run:

```
DEBUG="fly*" fly ...
```

and see if you notice anything funny in the logs.

**Comment by tvanier** [comment]
Thanks, running debug shows the potential problem is not in `Fly.prototype.source` but in `Fly.prototype.target`, where some path items seems to be filtered out. So `copy_images` will copy `images/icon.png` to `dist/icon.png`, not `dist/images/icon.png`. This naive change works for me:

``` diff
diff --git a/src/fly.js b/src/fly.js
index 6740c66..9450bc9 100644
--- a/src/fly.js
+++ b/src/fly.js
@@ -5,7 +5,7 @@ import mkdirp from "mkdirp"
 import chokidar from "chokidar"
 import Emitter from "./emitter"
 import Cat from "concat-with-sourcemaps"
-import { dirname, join, parse, sep } from "path"
+import { dirname, join, parse } from "path"
 import { readFile, writeFile, appendFile } from "mz/fs"
 import { log, alert, error, defer, flatten, expand } from "fly-util"
 const clear = defer(rimraf)
@@ -176,8 +176,7 @@ export default class Fly extends Emitter {
             this._.cat.add(`${base}`, data, map)
           } else {
             yield resolve(dirs, {
-              data, base: join(...parse(file).dir.split(sep)
-                .filter((path) => !~glob.split(sep).indexOf(path)),
+              data, base: join(parse(file).dir,
                 `${parse(file).name}${ext}`),
               map: this._.cat && this._.cat.sourceMap
             })
```

**Comment by lukeed** [comment]
Closed via #135

**Comment by ghost** [comment]
:ok_hand:
